firebase-firestore/src/main/java/com/google/firebase/firestore/core/SyncEngine.java

```diff
@@ -708,15 +710,15 @@ private void pumpEnqueuedLimboResolutions() {
   }
 
   @VisibleForTesting
-  public Map<DocumentKey, Integer> getActiveLimboDocumentResolutions() {
+  public HashMap<DocumentKey, Integer> getActiveLimboDocumentResolutions() {
     // Make a defensive copy as the Map continues to be modified.
-    return new HashMap<>(activeLimboTargetsByKey);
+    return new HashMap(activeLimboTargetsByKey);
   }
 
   @VisibleForTesting
-  public Queue<DocumentKey> getEnqueuedLimboDocumentResolutions() {
-    // Make a defensive copy as the Queue continues to be modified.
-    return new ArrayDeque<>(enqueuedLimboResolutions);
+  public LinkedHashSet<DocumentKey> getEnqueuedLimboDocumentResolutions() {
```

Comment:
Super nit (here and above): We usually try to use more generic types in our return types, as this allows us to change the implementation without changing the callsites. If the callsites don't require the `HashMap` or `LinkedHashSet` functionality, I would just use `Map<>` and `Set<>`. 

Please note that this might not apply if `getEnqueuedLimboDocumentResolutions()` only uses the Set API but its consumers rely on the iteration order of a LinkedHashSet. Then I would still return `LinkedHashSet` as we will not be able to replace it later. You should still generify the signature in  line 713 though.

Reply:
Done.